finesse/hardware/dummy_tc4820.py

```diff
@@ -0,0 +1,79 @@
+"""Provides a dummy TC4820 device."""
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass
+from decimal import Decimal
+from typing import Optional
+
+from .noise_producer import NoiseProducer
+from .tc4820_base import TC4820Base
+
+
+@dataclass
+class NoiseParameters:
+    """A compact way of expressing arguments to NoiseProducer."""
+
+    mean: float = 0.0
+    standard_deviation: float = 1.0
+    seed: Optional[int] = 42
+
+
+class DummyTC4820(TC4820Base):
+    """A dummy TC4820 device which produces random noise for its properties."""
+
+    def __init__(
+        self,
+        temperature_params: NoiseParameters = NoiseParameters(35.0, 2.0),
+        power_params: NoiseParameters = NoiseParameters(40.0, 2.0),
+        alarm_status: int = 0,
+        initial_set_point: Decimal = Decimal(70),
+    ) -> None:
```

Comment:
You're still totally breaking the API for TC4820-type devices. This cannot be used in place of the real TC4820, which is the whole point of mock devices. Who might want to customise the way you are enabling and why?

Reply:
It's not supposed to be a mock device for using in unit tests. It's purely to help us develop the GUI without access to the real hardware, as I did with SPiQE Record (which just generated sine waves for each of the channels). We can use unit tests + `pytest-qt` to check that things are working as expected, but it would still be helpful to be able to see whether the GUI is doing what we expect it to.

The API *is* the same for the two classes in the ways that matter: it has the same properties and because it uses `pubsub` in the same way it will look the same to the frontend. Yes, the constructors are different, but the `TC4820` constructor takes a `Serial` object as an argument, which we really don't want to have to mock up ourselves. I was planning to hide the dummy TC4820 functionality behind a developer-only option.

With SPiQE Record my dummy device was a bit lower-level and produced the same patterns of raw bytes as the real device, but in retrospect that was overkill and it would have been fine to just generate arrays of numbers, which is all the frontend required. The protocol for the TC4820 is a lot more complicated, so I don't want to have to implement an encoder for it if possible!

Reply:
I didn't hardcode the parameters because I thought it was bad practice, but we also probably want to use different parameters for the two TC4820 devices that will be connected.